### ISSUE TITLE
Why does setTimeout and setInterval can't be transpiled to Lua?

### ISSUE THREAD

**Issue by ProgrammingLife** [Perryvw/TypeScriptToLua#1467]
I've just noticed that setTimeout and setInterval can't be transpiled to Lua with TypescriptToLua but there are some solutions for this in Lua such as timers (wait), coroutines, etc... then why it can't transpile setTimeout and setInterval functions?

**Comment by lolleko** [comment]
Just checked, there is indeed no good way to get the current time. `os.time()` returns time and seconds only. On top of that it is system dependent und inaccurate.

So I don't think adding support for setInterval and setTimeout will be possible.

**Comment by Cheatoid** [comment]
One way would be to go full abstract/agnostic, so no matter what Lua environment is being used, a programmer would provide these requirements/arguments to the lualib. Expose a method to set it up before being able to use it?
Then the rest simply becomes just a common Timer implementation.

**Comment by Z3rio** [comment]
Yeah, some way of integrating this would be great, personally, I'm running a Lua runtime that does support this, which makes it a bit annoying that I cant use it.

**Comment by Perryvw** [comment]
If you have to provide some binding code to make setTimeout work with your own timers, what is the benefit over just using that timer declaration directly?

In fact you could just add your own setTimeout implementation to your environment to be used by the default declarations:
```ts
function setTimeout(handler: () => void, timeout: number) {
    // code calling your environment timer here
}
```
(you will need to make sure this lua file is loaded as part of the global scope ofc

**Comment by admiralnelson** [comment]
I implemented my own  setTimeout and setInterval https://github.com/admiralnelson/warhammer3-typescript-framework/blob/cdcd2a041abeea69bb7f4df72882523d53cb9ef5/template/campaign/mod/_MyProjectUtils.ts#L285
I guess it's because the implementation of setTimeout and setInterval depends on the platform you are working on, some might doesn't even have event loop at all.
